aten/src/ATen/native/cpu/UnaryOpsKernel.cpp

```diff
@@ -0,0 +1,132 @@
+#include "ATen/native/cpu/UnaryOpsKernel.h"
+#include "ATen/Dispatch.h"
+#include "ATen/Parallel.h"
+#include <iostream>
+#include <cmath>
+#include "ATen/native/cpu/Vec256.h"
+
+namespace at {
+namespace native {
+
+using namespace vec256;
+
+// This modifies arr in place with given OP
+template <class scalar_t, template <class> class VOP, template <class> class SOP, CPUCapability C>
+inline void kernel_(scalar_t *arr_out, const scalar_t *arr_in, size_t start, size_t end) {
+  // Use all 16 registers.
```

Comment:
This looks more complicated than necessary. I don't think there's any advantage to using more registers than necessary. Just do:

```
for (; k < (end - start) / width; k++) {
  Vec<scalar_t> a;
  a.load(...);
  a = VOP<scalar_t>()(a);
  a.store(...);
}
...
```

Reply:
I'll measure this.